packages/flutter_tools/lib/src/flx.dart

```diff
@@ -235,7 +230,41 @@ Future<int> build(
       return result;
     }
 
-    archive.addFile(_createSnapshotFile(snapshotPath));
+    snapshotFile = _createSnapshotFile(snapshotPath);
+  }
+
+  return assemble(
+      manifestDescriptor: manifestDescriptor,
+      snapshotFile: snapshotFile,
+      assetBasePath: assetBasePath,
+      materialAssetBasePath: materialAssetBasePath,
+      outputPath: outputPath,
+      privateKeyPath: privateKeyPath
+  );
+}
+
+/// Assembles a Flutter .flx file from a pre-existing manifest descriptor
+/// and a pre-compiled snapshot.
+///
+/// This may be called by external build toolchains, so practice caution
+/// when changing this method signature (alert flutter-dev).
+Future<int> assemble({
```

Comment:
Sounds good. If you're using the API, we want to make refactorings of the public API much more mindful, which I think means promoting the public api bits to `lib/`.